resources/prosody-plugins/token/util.lib.lua

```diff
@@ -112,6 +112,14 @@ function Util:get_public_key(keyId)
     if content == nil then
         -- If the key is not found in the cache.
         module:log("debug", "Cache miss for key: "..keyId);
+        
+        local external_public_key = prosody.events.fire_event("pre-jitsi-authentication-public-key-fetch", keyId);
```

Comment:
I disagree with this implementation, as it allows for future use of this kid for ANY requests.  Instead, we should be providing the external public key on the session before this step, via a hook in mod_auth_token.  Then when we get to the step of attempting to get the public key, we can check for it on the session and use it if it's present, without even attempting to execute get_public_key.